polyfills/Array/from/polyfill.js

```diff
@@ -20,6 +61,34 @@ Object.defineProperty(Array, 'from', {
 		length = Math.min(Math.max(Number(arraylike.length) || 0, 0), 9007199254740991),
 		value;
 
+		// variables for rebuilding array from iterator
+		var arrayFromIterable;
+
+		// if it is an iterable treat like one
+		// it is an iterable if 'next' is a function but it has not been defined on
+		// the object itself.
+		if (typeof arraylike.next === 'function' && arraylike.hasOwnProperty('next') === false) {
+			arrayFromIterable = parseIterable(arraylike);
+		}
+
+		//if it is a Map or a Set then handle them appropriately
+		if (
+			typeof arraylike.entries === 'function' &&
+			typeof arraylike.values === 'function'
+		) {
+			if (arraylike.constructor.name === 'Set') {
+				arrayFromIterable = parseIterable(arraylike.values());
+			}
+			if (arraylike.constructor.name === 'Map') {
+				arrayFromIterable = parseIterable(arraylike.entries());
+			}
+		}
+
+		if (arrayFromIterable) {
+			arraylike = arrayFromIterable;
```

Comment:
This looks to me like it is iterating through the arrayLike object once if it does not have an iterable interface and twice if it does have one.